src/CreditCard.java

```diff
@@ -0,0 +1,71 @@
+import java.math.BigDecimal;
+import java.time.LocalDate;
+
+public class CreditCard extends PaymentMethod {
+
+    private String cardNumber;
+    private LocalDate expDate;
+    private int cvv;
+    private String cardName;
+    private Integer flexPaymentNumber;
+
+
+    public CreditCard() {
+    }
+
+    public CreditCard(String cardNumber, LocalDate expDate, int cvv, String cardName, Integer flexPaymentNumber) {
+        this.cardNumber = cardNumber;
+        this.expDate = expDate;
+        this.cvv = cvv;
+        this.cardName = cardName;
+        this.flexPaymentNumber = flexPaymentNumber;
+    }
+
+
+    public String getCardNumber() {
+        return cardNumber;
+    }
+
+    public LocalDate getExpDate() throws ExpDateException{
+        if (expDate == null){
+            throw new ExpDateException("O cartão não pode ter a data de validade nula.");
+        }
+        if (expDate.isBefore(LocalDate.now())){
+            throw new ExpDateException("O cartão está com a validade expirada");
+        }
```

Comment:
Essa validação deveria estar no getter mesmo?

Reply:
Penso que não faz muito sentido cadastrar um cartão com uma data errada, mas se a pessoa tem o cartão cadastrado já e vai usar o cartão, o que tenho que verificar é quando for pegar esse cartão ver se a data está válida naquele momento. Acha melhor colocar no setter também?

Reply:
Colocar validações no Getter é uma má pratica. Nesse cenário que você comentou, você teria que verificar no momento do "checkout" do pagamento se aquele cartão ainda é válido, isso assumindo que o cartão ficou armazenado em algum lugar.

Se o seu getter ja joga uma exception, nenhum lugar do seu código vai conseguir dar um get na data de validade do cartão para verificar isso.

Reply:
Entendi. Então vou deixar só no setter, no momento. Pode ser?